optaplanner-core/src/test/java/org/optaplanner/core/impl/heuristic/selector/entity/pillar/DefaultPillarSelectorTest.java

```diff
@@ -226,11 +226,11 @@ public void randomWithSubs() {
         // nextInt pattern: pillarIndex, subPillarSize, element 0, element 1, element 2, ...
         // Expected pillar cache: [a], [b, d], [c, e, f]
         when(workingRandom.nextInt(anyInt())).thenReturn(
-                0, 0, 0, // [a]
-                2, 1, 0, 0, // [c, e, f]
```

Comment:
A suggestion to make the test more readable:
would you please explain on one example, what is the relation of mocked random numbers to the subpillar being selected? Even with the comment on lines 226 and 227 it's difficult to see the connection without detailed knowledge of the implementation.

Reply:
This part of the test is painful indeed, in the original code as much as it is now.
It's mocking random numbers to trigger certain selections.

But this is the lesser evil.
The other 2 options I am aware of are:
- not testing it (or not testing parts of it)
- statistical testing that randomly fails sometimes (we don't have the setup to accommodate this)
So, we stick with this approach until something better comes along.

Reply:
@triceo @triceo Just thinking further on this (already merged, so a different issue):
What if we do statistical testing with a fixed random seed? That means we don't have random failing tests.

For example:
Given 4 entities, we select 100 subpillars and we validate that every combination of those 4 is selected.

Reply:
I'm not sure, @ge0ffrey. It will make the tests run much longer, and I have no idea how to pick a random number and a data set that will cause it to trigger exactly what we need.

Too much effort for too little value?

Reply:
Let's just provide detailed explanation of what's happening with random numbers in the test and it will be much better than what we have now.